packages/shared/src/open-feature.ts

```diff
@@ -23,10 +27,10 @@ export abstract class OpenFeatureCommonAPI<P extends CommonProvider = CommonProv
 
   protected abstract _defaultProvider: P;
 
-  private readonly _events = new OpenFeatureEventEmitter(() => this._logger);
-  private readonly _clientEventHandlers: Map<string | undefined, [ProviderEvents, EventHandler][]> = new Map();
+  private readonly _events = new InternalEventEmitter(() => this._logger);
+  private readonly _clientEventHandlers: Map<string | undefined, [ProviderEvents, EventHandler<any>][]> = new Map();
```

Comment:
```suggestion
  private readonly _clientEventHandlers: Map<string | undefined, [ProviderEvents, EventHandler<ProviderEvents>][]> = new Map();
```

Here we can use ProviderEvents instead of any as we are never using narrower types here.
Thats not possible in the OpenFeatureEvenetEmitter, as we use the narrower types there ( https://github.com/open-feature/js-sdk/pull/490/files#diff-92bf71360d9fdb080cd82b4e1e5340f32ba26fe878ffc9529f87f32cf629931eR9)

Reply:
Good catch!